### PR TITLE
nixos/openwebrx: add more codecs

### PR DESCRIPTION
**Pull request by astro**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/codec2/default.nix [hunk 1]
```diff
@@ -13,6 +13,12 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ cmake ];
 
+  # Install a binary that is used by openwebrx
+  postInstall = ''
+    mkdir -p $out/bin
+    install -m 0755 src/freedv_rx $out/bin/
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/codec2/default.nix:19]:
```suggestion
    install -Dm0755 src/freedv_rx $out/bin/
```

File: pkgs/applications/radio/m17-cxx-demod/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  nativeBuildInputs = [ cmake pkg-config boost ];
+  buildInputs = [ codec2 ];
+
+  meta = with lib; {
+    description = "M17 Demodulator in C++";
+    homepage = "https://github.com/mobilinkd/m17-cxx-demod";
+    license = licenses.gpl3Only;
+    platforms = platforms.unix;
+    maintainers = with maintainers; [ astro ];
```
Comment (SuperSandro2000) [inline pkgs/applications/radio/m17-cxx-demod/default.nix:22]: We could also put that under c3d2 if you like.

**Review by SuperSandro2000 — COMMENTED** [review]
otherwise LGTM

File: pkgs/applications/radio/m17-cxx-demod/default.nix (hunk 2 above)
Comment (markuskowa) [reply pkgs/applications/radio/m17-cxx-demod/default.nix:22]: What does "c3d2" mean?

Comment (astro) [reply pkgs/applications/radio/m17-cxx-demod/default.nix:22]: We collectively maintain packages that are used in our shared infrastructure. See #178612

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/development/libraries/codec2/default.nix [hunk 3]
```diff
@@ -13,6 +13,11 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ cmake ];
 
+  # Install a binary that is used by openwebrx
+  postInstall = ''
+    install -Dm0755 src/freedv_rx $out/bin/
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/codec2/default.nix:18]:
```suggestion
    install -Dm0755 src/freedv_rx -t $out/bin/
```

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Review by markuskowa — APPROVED** [review]

**Pull request merged**